cephadm-preflight.yml

```
@@ -48,6 +48,224 @@
     - name: redhat family of OS related tasks
       when: ansible_facts['os_family'] == 'RedHat'
       block:
+        - name: Initialize os_check variable
+          set_fact:
+            os_check: { "failed": false }
+
+        - name: OS Preflight Checks
+          block:
+            - name: Initialize check results
+              set_fact:
+                preflight_results: []
+
+            - name: Ensure OS is RHEL 9+
+              fail:
+                msg: "Ceph requires RHEL 9+. Detected: {{ansible_facts['distribution']}} {{ansible_facts['distribution_version']}}"
+              when: 
+                - ansible_facts['distribution'] not in ['RedHat'] or ansible_facts['distribution_major_version'] | int <9
+              register: os_check
+              ignore_errors: true
+
+            - name: Store OS check result
+              set_fact:
+                preflight_results: "{{ preflight_results + [{'Check': 'OS Version', 'Result': 'PASS' if ansible_facts['distribution_major_version'] | int >= 9 else 'FAIL'}] }}"
+            
+            - name: Check if SELinux is in Enforcing Mode
+              command: getenforce
+              register: selinux_status
+              changed_when: false
+              ignore_errors: true
```

I would strongly advise against using `ignore_errors: true`.

Wouldn't something like the following be better?

```suggestion
            - name: Check if SELinux is in Enforcing Mode
              ansible.posix.selinux:
                policy: targeted
              register: selinux_status
              changed_when: false
```

---

Using the above suggestion to set the SELinux.